src/swapd/runtime.rs

```diff
@@ -462,82 +472,20 @@ impl Runtime {
         )?)
     }
 
-    /// Checkpoint the state before executing the buy procedure signature
-    fn try_checkpoint_bob_pre_buy(&mut self, endpoints: &mut Endpoints) -> Result<(), Error> {
-        if let Some(buy_proc) = self.state.get_bob_buy_proc() {
-            if self.txs.contains_key(&TxLabel::Cancel)
-                && self.txs.contains_key(&TxLabel::Refund)
-                && self.state.b_sup_checkpoint_pre_buy()
-            {
-                debug!("{} | checkpointing bob pre buy swapd state", self.swap_id);
-                checkpoint_send(
-                    endpoints,
-                    self.swap_id,
-                    self.identity(),
-                    ServiceId::Database,
-                    CheckpointState::CheckpointSwapd(CheckpointSwapd {
-                        state: self.state.clone(),
-                        last_msg: PeerMsg::BuyProcedureSignature(buy_proc),
-                        enquirer: self.enquirer.clone(),
-                        temporal_safety: self.temporal_safety.clone(),
-                        txs: self.txs.clone().drain().collect(),
-                        txids: self.syncer_state.tasks.txids.clone().drain().collect(),
-                        pending_requests: self.pending_requests().clone().drain().collect(),
-                        pending_broadcasts: self.syncer_state.pending_broadcast_txs(),
-                        xmr_addr_addendum: self.syncer_state.xmr_addr_addendum.clone(),
-                        local_trade_role: self.local_trade_role,
-                        connected_counterparty_node_id: get_node_id(&self.peer_service),
-                        public_offer: self.public_offer.clone(),
-                        monero_address_creation_height: self.monero_address_creation_height.clone(),
-                    }),
-                )?;
-            } else {
-                debug!(
-                    "{} | trying checkpoint Bob pre buy but txs not received yet",
-                    self.swap_id,
-                );
-                if !self.state.get_is_checkpoint_pre_buy_ready() {
-                    warn!(
-                        "{} | trying checkpoint Bob pre buy but incorrect state: {}",
-                        self.swap_id, self.state,
-                    );
-                }
-            }
-        } else {
-            debug!(
-                "{} | trying checkpoint Bob pre buy but Buy Procedure Signature not received yet",
-                self.swap_id,
-            );
-        }
-        Ok(())
-    }
-
-    fn handle_msg(
+    pub fn handle_msg(
```

Comment:
assuming was only there for iterating fast and can be reverted
```suggestion
    fn handle_msg(
```

Reply:
25d2e10f325e6b5bf188d616c8b85ea7d242b4cb